### PR TITLE
Update doc for dynamic parameters supporting array

### PR DESCRIPTION
**Pull request by sreemanamala**
### Description
Update dynamic parameter docs to provide how it can used to replace an Array

<hr>

##### Key changed/added classes in this PR
 * `sql.md`
 * `sql-api.md`
 * `sql-jdbc.md`

<hr>

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: docs/querying/sql.md [hunk 1]
```diff
@@ -409,3 +409,27 @@ To solve this issue, explicitly provide the type of the dynamic parameter using
 ```
 SELECT * FROM druid.foo WHERE dim1 like CONCAT('%', CAST (? AS VARCHAR), '%')
 ```
+
+With the support of dynamic parameters being extended to arrays, an array with a huge number of entries can be
```
[the remark this answers is not in the document]
Comment (sreemanamala) [reply docs/querying/sql.md:413]: It sounds the same for me after you pointing out. It was done as part of Druid 30. will remove `being` from the line. Thanks!

File: docs/querying/sql.md [hunk 2]
```diff
@@ -409,3 +409,27 @@ To solve this issue, explicitly provide the type of the dynamic parameter using
 ```
 SELECT * FROM druid.foo WHERE dim1 like CONCAT('%', CAST (? AS VARCHAR), '%')
 ```
+
+With the support of dynamic parameters extended to arrays, an array with a huge number of entries can be
+programmatically supplied through a parameter.
```
Comment (abhishekagarwal87) [inline docs/querying/sql.md:414]: its phrased as an announcement (ok for release notes but not for the docs). 
```suggestion
Arrays can also be passed as dynamic parameters. 
```

File: docs/querying/sql.md [hunk 3]
```diff
@@ -409,3 +409,26 @@ To solve this issue, explicitly provide the type of the dynamic parameter using
...
 SELECT * FROM druid.foo WHERE dim1 like CONCAT('%', CAST (? AS VARCHAR), '%')
 ```
+
+Arrays can also be passed as dynamic parameters.
+
+for example:
+```json
+{
+    "query": "SELECT doubleArrayColumn from druid.table where ARRAY_CONTAINS(?, doubleArrayColumn)",
+    "Parameters":[
```
Comment (kfaraz) [inline docs/querying/sql.md:419]: JSON field names are case-sensitive. Same comment in the other occurrence.

```suggestion
    "parameters":[
```

File: docs/querying/sql.md [hunk 4]
```diff
@@ -409,3 +409,26 @@ To solve this issue, explicitly provide the type of the dynamic parameter using
...
 SELECT * FROM druid.foo WHERE dim1 like CONCAT('%', CAST (? AS VARCHAR), '%')
 ```
+
+Dynamic parameters can also replace arrays, reducing the parsing time.
+
+for example:
+```json
+{
+    "query": "SELECT doubleArrayColumn from druid.table where ARRAY_CONTAINS(?, doubleArrayColumn)",
+    "parameters":[
```
Comment (asdf2014) [inline docs/querying/sql.md:419]:
```suggestion
    "parameters": [
```

File: docs/querying/sql.md [hunk 5]
```diff
@@ -409,3 +409,26 @@ To solve this issue, explicitly provide the type of the dynamic parameter using
...
+      {"type":"ARRAY", "value":[-25.7, null, 36.85]}
+    ]
+}
+```
+
+Also, an IN filter being supplied with a lot of values, can be replaced by a dynamic parameter passed inside [SCALAR_IN_ARRAY](sql-functions.md#scalar_in_array)
+```json
+{
+    "query": "SELECT count(city) from druid.table where SCALAR_IN_ARRAY(city, ?)",
+    "parameters":[
```
Comment (asdf2014) [inline docs/querying/sql.md:429]:
```suggestion
    "parameters": [
```

File: docs/querying/sql.md [hunk 6]
```diff
@@ -409,3 +409,26 @@ To solve this issue, explicitly provide the type of the dynamic parameter using
...
+
+for example:
+```json
+{
+    "query": "SELECT doubleArrayColumn from druid.table where ARRAY_CONTAINS(?, doubleArrayColumn)",
+    "parameters": [
+      {"type":"ARRAY", "value":[-25.7, null, 36.85]}
+    ]
+}
+```
```
Comment (kgyrtkirk) [inline docs/querying/sql.md:423]: I feel like this is `sql-api` specific so it shouldn't it be inside `../api-reference/sql-api.md` ?

File: docs/querying/sql.md [hunk 7]
```diff
@@ -409,3 +409,26 @@ To solve this issue, explicitly provide the type of the dynamic parameter using
...
+
+Also, an IN filter being supplied with a lot of values, can be replaced by a dynamic parameter passed inside [SCALAR_IN_ARRAY](sql-functions.md#scalar_in_array)
+```json
+{
+    "query": "SELECT count(city) from druid.table where SCALAR_IN_ARRAY(city, ?)",
+    "parameters": [
+      {"type":"ARRAY", "value":["Vienna", "Seoul", "San Francisco"]}
+    ]
+}
+```
```
Comment (kgyrtkirk) [inline docs/querying/sql.md:433]: I've played with similar things for jdbc api earlier; it seemed a bit harder to use it from there
should we include those in `sql-api.md`?

```java

import com.google.common.base.Joiner;
import com.google.common.collect.ImmutableList;
import org.apache.calcite.avatica.ColumnMetaData;
import org.apache.calcite.avatica.SqlType;
import org.apache.calcite.avatica.ColumnMetaData.AvaticaType;
import org.apache.calcite.avatica.ColumnMetaData.Rep;
import org.apache.calcite.avatica.util.ArrayFactoryImpl;
import org.junit.jupiter.api.Test;

import java.sql.Array;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Types;
import java.util.List;
import java.util.TimeZone;

public class Asd
{
  @Test
  public void scalarInArrayWithStringToArray() throws SQLException
  {
    String connectionURL = "druidtest:///";
    Connection connection = DriverManager.getConnection(connectionURL);
    PreparedStatement stmt = connection
        .prepareStatement("select l1 from numfoo where SCALAR_IN_ARRAY(l1, STRING_TO_ARRAY(CAST(? as varchar),','))");
    List<Integer> li = ImmutableList.of(0, 7);
    String sqlArg = Joiner.on(",").join(li);
    stmt.setString(1, sqlArg);
    stmt.executeQuery();
    ResultSet rs = stmt.getResultSet();
    while (rs.next()) {
      System.out.println(rs.getObject(1));
    }
  }

  @Test
  public void scalarInArrayNativeArray() throws SQLException
  {
    String connectionURL = "druidtest:///";
    Connection connection = DriverManager.getConnection(connectionURL);
    PreparedStatement stmt = connection.prepareStatement("select l1 from numfoo where SCALAR_IN_ARRAY(l1, ?)");
    Iterable<Object> li = ImmutableList.of(0, 7);
    ArrayFactoryImpl a = new ArrayFactoryImpl(TimeZone.getDefault());
    org.apache.calcite.avatica.proto.Common.AvaticaType proto;
    AvaticaType type = ColumnMetaData.scalar(Types.INTEGER, SqlType.INTEGER.name(), Rep.INTEGER);
    Array arr = a.createArray(type, li);
    stmt.setArray(1, arr);
    stmt.executeQuery();
    ResultSet rs = stmt.getResultSet();
    while (rs.next()) {
      System.out.println(rs.getObject(1));
    }
  }
}
```

Comment (sreemanamala) [reply docs/querying/sql.md:423]: sample code added to `sql-jdbc.md`

File: docs/querying/sql.md (hunk 6 above)
Comment (sreemanamala) [reply docs/querying/sql.md]: made these simple sql queries and provided the example parameters inside `sql-api.md`

**Review by kgyrtkirk — APPROVED** [review]
+1

**Pull request merged**